### PR TITLE
chore(primitives): remove optimism field in ChainSpec

### PR DESCRIPTION
**Pull request by merklefruit**
- addresses https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376101312
- addresses https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376103265

### CODE REVIEW

File: crates/primitives/src/chain/mod.rs [hunk 1]
```diff
@@ -151,6 +147,18 @@ impl Chain {
...
+    pub fn is_optimism(self) -> bool {
+        matches!(
+            self,
+            Chain::Named(NamedChain::Optimism) |
+                Chain::Named(NamedChain::OptimismGoerli) |
+                Chain::Named(NamedChain::OptimismKovan) |
+                Chain::Named(NamedChain::Base) |
+                Chain::Named(NamedChain::BaseGoerli)
+        )
+    }
```
Comment (merklefruit) [inline crates/primitives/src/chain/mod.rs:160]: Don't really like this, feel like having the `optimism` bool flag was cleaner, wdyt @refcell?

Comment (clabby) [reply crates/primitives/src/chain/mod.rs]: I also preferred the `optimism` bool flag here, felt more simple. Allows for easy additions of OP devnet chain configs etc.

Comment (refcell) [reply crates/primitives/src/chain/mod.rs]: Agreed, prefer to keep it simple - we can push back on this :)

Comment (clabby) [reply crates/primitives/src/chain/mod.rs]: Did matt explicitly ask us to remove the `optimism` bool?

Comment (merklefruit) [reply crates/primitives/src/chain/mod.rs]: related comment: https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376103265

Comment (refcell) [reply crates/primitives/src/chain/mod.rs]: [Replied](https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376936233) in the thread asking to keep the `optimism` bool flag.

File: crates/consensus/common/src/validation.rs [hunk 2]
```diff
@@ -300,33 +297,33 @@ pub fn validate_header_regarding_parent(
...
-
-    #[cfg(feature = "optimism")]
-    if !chain_spec.optimism {
-        check_gas_limit(parent, child, chain_spec)?;
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "optimism")] {
+            // On Optimism, the gas limit can adjust instantly, so we skip this check
+        } else {
+            check_gas_limit(parent, child, chain_spec)?;
+        }
```
Comment (refcell) [inline crates/consensus/common/src/validation.rs:326]: We want to still check if the chain is `optimism` even if reth is compiled with the `optimism` feature flag on.

Think this should be
```suggestion
    cfg_if::cfg_if! {
        if #[cfg(feature = "optimism")] {
            // On Optimism, the gas limit can adjust instantly, so we skip this check
            if !chain_spec.is_optimism() {
                check_gas_limit(parent, child, chain_spec)?;
            }
        } else {
            check_gas_limit(parent, child, chain_spec)?;
        }
```

**Review by refcell — CHANGES_REQUESTED** [review]
One small change, otherwise looks good!

**Review by refcell — APPROVED** [review]

**Pull request merged**